controllers/main.go

```diff
@@ -185,6 +206,39 @@ func (controller *MainController) getNetworkName() string {
 	return controller.params.Name
 }
 
+// agendas() returns agendas and their statuses. Fetches agenda status from
+// dcrdata.org if past agenda.Timer limit from previous fetch. Caches agenda
+// data for agendasCacheLife. This method is safe for concurrent use.
+func (controller *MainController) agendas() []agenda {
+	defer agendasCache.Unlock()
+	agendasCache.Lock()
+	now := time.Now()
+	if agendasCache.timer.After(now) {
+		return *agendasCache.agendas
+	}
+	url := fmt.Sprintf("https://%s.dcrdata.org/api/agendas", controller.getNetworkName())
+	agendaInfos, err := dcrDataAgendas(url)
+	if err != nil {
+		log.Warnf("unable to retrieve data from %v: %v", url, err)
```

Comment:
With the way this is currently written, a failed call to dcrdata will result in the cached data being overwritten with an empty array, and then no further attempts to contact dcrdata will occur until the timeout has elapsed. If an error occurs we should probably return the old cached data after logging the warning. That way, next time `agendas()` is invoked, the call to dcrdata will be retried.